airflow/providers/amazon/aws/operators/redshift.py

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import List, Optional, Union
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.redshift import RedshiftSQLHook
+
+
+class RedshiftSQLOperator(BaseOperator):
+    """
+    Executes SQL Statements against an Amazon Redshift cluster
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:RedshiftSQLOperator`
+
+    :param sql: the sql code to be executed
+    :type sql: Can receive a str representing a sql statement,
+        a list of str (sql statements)
+    :param redshift_conn_id: reference to
+        :ref:`Amazon Redshift connection id<howto/connection:redshift>`
+    :type redshift_conn_id: str
+    :param parameters: (optional) the parameters to render the SQL query with.
+    :type parameters: dict or iterable
+    :param autocommit: if True, each command is automatically committed.
+        (default value: False)
+    :type autocommit: bool
+    """
+
+    template_fields = ('sql',)
+    template_ext = ('.sql',)
+
+    def __init__(
+        self,
+        *,
+        sql: Union[str, List[str]],
+        redshift_conn_id: str = 'redshift_default',
```

Comment:
Knowing that it's been the pattern in operators for a long time but it seems cleaner to not hardcode the default value for `redshift_conn_id` but use `RedshiftSQLHook.defaul_conn_name`.

Reply:
@ashb you have a take on this?  if you DRY it, then i think you won't see what the default is in IDE

Reply:
I don't have much opinion, bit I don't really see what adding the default as a class property gives us.

Reply:
removed hardcoded value in a02cdbc

Reply:
> I don't have much opinion, bit I don't really see what adding the default as a class property gives us.

The class property thing is something defined in DbApiHook, just a way of abstracting common hook init boilerplate, but I'd agree it's probably not that helpful.

Anyway, just cus the hook default conn id is `redshift_default`, there's no requirement that the operator has to use the same default.  So I think it's reasonable to duplicate the string default `redshift_conn_id='redshift_default'`.  And I think as a user it is more transparent and therefore easier to use.  But I'll leave it up to you @Brooke-white since you already made the change whether you want to leave it or put the literal back.

Reply:
reverted this change in 005531b, I agree this will be more transparent.